nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -581,30 +585,31 @@ private Status getStatus(final Logger logger) {
         return new Status(port, pid, pingSuccess, alive);
     }
 
-    public void status() throws IOException {
+    public int status() throws IOException {
         final Logger logger = cmdLogger;
         final Status status = getStatus(logger);
         if (status.isRespondingToPing()) {
             logger.info("Apache NiFi is currently running, listening to Bootstrap on port {}, PID={}",
                     new Object[]{status.getPort(), status.getPid() == null ? "unknown" : status.getPid()});
-            return;
+            return 0;
         }
 
         if (status.isProcessRunning()) {
             logger.info("Apache NiFi is running at PID {} but is not responding to ping requests", status.getPid());
-            return;
+            return 0;
```

Comment:
I do not mind if you change the value in this case. You may take the changes and commit them modified, no problem to me. Your proposal is cleaner in semantic sense, it may cause unnecessary process killing, but it well may be desired behavior in NiFi's case - if its not responding == its dead. So lets do it. I am ok with it.